app/connectors/PaperlessPreferenceConnector.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import com.google.inject.{ImplementedBy, Inject}
+import config.FrontendAppConfig
+import models.paperless.PaperlessStatusResponse
+import play.api.Logger
+import play.api.mvc.{AnyContent, Request}
+import uk.gov.hmrc.crypto.{ApplicationCrypto, PlainText}
+import uk.gov.hmrc.http.HttpReads.Implicits._
+import uk.gov.hmrc.play.bootstrap.http.HttpClient
+import uk.gov.hmrc.play.partials.HeaderCarrierForPartialsConverter
+import views.html.helper.urlEncode
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class PaperlessPreferenceConnectorImpl @Inject()(
+                                                  appConfig: FrontendAppConfig,
+                                                  httpClient: HttpClient,
+                                                  applicationCrypto:  ApplicationCrypto,
+                                                )
+  extends PaperlessPreferenceConnector with HeaderCarrierForPartialsConverter {
+
+  override def crypto: String => String = cookie => cookie
+
+  def getPaperlessStatus(returnUrl: String)(implicit request: Request[AnyContent], ec: ExecutionContext): Future[Either[String, PaperlessStatusResponse]] = {
+    val paperlessStatusUrl =
+      s"${appConfig.preferencesFrontendHost}/paperless/status" +
+        s"?returnUrl=${ encryptAndEncode( returnUrl ) }" +
+        s"&returnLinkText=${ encryptAndEncode( "Continue" ) }"
+
+    Logger.debug(s"[PaperlessPreferenceConnector][getPaperlessPreference] $paperlessStatusUrl")
```

Comment:
Do we need this?

Reply:
It doesn't harm and we do not have DEBUG level set in prod environments

Reply:
think about this in context: 
1. you are calling paperless/status endpoint to get a paperless status;
2. the possible return values are: https://github.com/hmrc/preferences-frontend/blob/master/app/model/StatusWithUrl.scala#L16
3. You have 4 paperless statuses which indicate a problem: PAPER, NEW_CUSTOMER, EMAIL_NOT_VERIFIED, EMAIL_BOUNCED;
4. You have 1 paperless status which indicates a customer is paperless with no problem: ALRIGHT;
5. Soon you will have 1 paperless status which will indicate a customer is paperless but it has a problem: OLD_VERSION, meaning he opted in with an old version of the terms and conditions and needs to RE_OPT_IN;

So in the above context, some clients, like mobile app want to have a fine grained status to display different screens/user journeys. For BTA they would want to show a specific message for the customers. PTA don't care as they just call paperless interrupt and we do the status calculation and displaying the page for them;
For other clients, like yourself, you would want to know just if the customer is paperless so that he can receive digital communication. For this use case we will have a field called isPaperless: Boolean. 
If you have any suggestion of a better alternative to ALRIGHT, we would be happy to hear it ;)